### PR TITLE
[MINOR][CORE] Process bar should print new line to avoid polluting logs

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

Use `println()` instead of `print()` to show process bar in console.

### Why are the changes needed?

Logs are polluted by process bar:

image

This is easy to reproduce:

1. start `./bin/spark-shell`
2. `sc.setLogLevel("INFO")`
3. run: `spark.range(100000000).coalesce(1).write.parquet("/tmp/result")`

### Does this PR introduce any user-facing change?

Yeah, more friendly format in console.

### How was this patch tested?

Tested manually.

### CODE REVIEW

**Comment by srowen** [comment]
You can disable it with `spark.ui.showConsoleProgress` in this case.
But adding a line feed will cause it to print each update on a newline, won't it? it's meant to print over itself. Did you mean to add one CR/LF before or after the loop?

**Comment by Ngone51** [comment]
> But adding a line feed will cause it to print each update on a newline, won't it? it's meant to print over itself.

Oh, yes, you're right. 

> Did you mean to add one CR/LF before or after the loop?

Ah, I don't understand well. But I just want to add a new line for following logs.

**Comment by srowen** [comment]
What if instead you `print(CR + bar + CR)`?

**Comment by Ngone51** [comment]
> What if instead you print(CR + bar + CR)?

This works after I quickly tried locally. But I didn't see progress bar move on(possibly due to the job is too small) to validate it still print over itself.

File: core/src/main/scala/org/apache/spark/ui/ConsoleProgressBar.scala [hunk 1]
```diff
@@ -94,7 +94,9 @@ private[spark] class ConsoleProgressBar(sc: SparkContext) extends Logging {
     // only refresh if it's changed OR after 1 minute (or the ssh connection will be closed
     // after idle some time)
     if (bar != lastProgressBar || now - lastUpdateTime > 60 * 1000L) {
-      System.err.print(CR + bar)
+      // scalastyle:off println
+      System.err.println(CR + bar)
```
Comment (HyukjinKwon) [inline core/src/main/scala/org/apache/spark/ui/ConsoleProgressBar.scala]: Hm? this will show every line for every time it's updated.

Comment (HyukjinKwon) [reply core/src/main/scala/org/apache/spark/ui/ConsoleProgressBar.scala]: Oh, okay, that's already pointed out at https://github.com/apache/spark/pull/27061#issuecomment-569943770

**Comment by srowen** [comment]
The change I suggested ought to be OK. Note that it does print a CR already when it is `clear()`ed, so i guess this happens while logs occur during progress? In that case the progress bar prints over the log I guess, but, that always happened.

It would be worth just making a bigger dummy test to verify it prints over itself as usual with that change, if you want to proceed.

**Comment by Ngone51** [comment]
@srowen I did another manually test by inserting/modifying some code into `ConsoleProgressBar` and the behavior looks good.

Did you ask for a unit test?

**Comment by srowen** [comment]
Jenkins retest this please

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by Ngone51** [comment]
Thanks!

**Pull request closed without merging**